### DISCUSSION TITLE
Martin v0.11.6 much faster than v0.12.0 and 0.13.0??

### DISCUSSION THREAD

**Discussion by radimkrchnacek** [maplibre/martin#1161] [Q&A]
I am testing 3 different versions of Martin on localhost with the same settings (default settings, just caching disabled) and the same postgis 3.3 database table. Versions v0.12.0 and v0.13.0 seem to be much slower than v0.11.6
I also tried running Martin with `RUST_LOG=debug` and the DB queries are identical for all 3 versions.
What might be the cause of these huge performance differences?

v0.12.0:
![martin-v0 12 0](https://github.com/maplibre/martin/assets/94595399/b0f1a597-9d11-4fbf-9aca-ea5c4061406e)

v0.11.6:
![martin-v0 11 6](https://github.com/maplibre/martin/assets/94595399/a0d98e2a-c05a-4d57-8d30-00af492867d3)

v.0.13.0
![martin-v0 13 0](https://github.com/maplibre/martin/assets/94595399/30a1043d-2a7e-4fa3-aca9-b1edea9ac360)

**Comment by nyurik** [comment]
@radimkrchnacek thanks for reporting this issue!   I did make some changes in the tile handling logic in order to support tile caching, and I was hoping it was a negligible change - so clearly should investigate this. I use `oha` tool to stress-test martin's response with a "null-source", but I don't think I ever ran the proper v0.11 vs later versions.

Other than this, it could be due to a dependency upgrade, e.g. newer version of actix, sqlx, or many other libs, but I doubt that's the case... Hopefully i will find some time to do a perf deep-dive to see what might have caused this... but as always - this is my volunteer project, so time constraints are ... what they are :)
